@vitejs/plugin-legacy fails to import >= 6.1.0

### Describe the bug

I'm trying to upgrade `@vitejs/plugin-legacy` from `6.0.2` to `6.1.0`.

What I expect is compilation to succeed using the syntax used previously or an alternative, if this would be a major version change.

What's actually is happening, is that `@vitejs/plugin-legacy` no longer has a default export and compilation fails after the upgrade with no other option.

### Reproduction

https://github.com/ViMaSter/gdqreminder-mobile/pull/1722/files

### Steps to reproduce

1. Clone the linked project's default branch (which uses `@vitejs/plugin-legacy` 6.0.2)
2. Prepare by running `nvm install && nvm use && npm install`
3. Run `npx vue-tsc --noEmit` which succeeds
4. Clone the linked project on the `dependabot/npm_and_yarn/vitejs/plugin-legacy-6.1.0` branch (which uses `@vitejs/plugin-legacy` 6.1.0)
5. Prepare by running `nvm install && nvm use && npm install`
6. Run `npx vue-tsc --noEmit` which fails
7. See this error: `vite.config.ts(3,8): error TS1192: Module '".../node_modules/@vitejs/plugin-legacy/dist/index"' has no default export.`

### System Info

```shell
System:
  OS: Windows 10 10.0.19045
  CPU: (16) x64 12th Gen Intel(R) Core(TM) i5-12600K
  Memory: 45.09 GB / 63.79 GB
Binaries:
  Node: 22.15.0 - C:\Program Files\nodejs\node.EXE
  npm: 10.9.2 - D:\prj\10_mobile\gdqreminder-mobile\node_modules\.bin\npm.CMD
  pnpm: 8.15.1 - ~\AppData\Local\pnpm\pnpm.EXE
Browsers:
  Edge: Chromium (127.0.2651.74)
  Internet Explorer: 11.0.19041.4355
npmPackages:
  @vitejs/plugin-legacy: ^6.1.0 => 6.1.0
  @vitejs/plugin-vue: ^5.2.3 => 5.2.3
  @vitejs/plugin-vue-jsx: ^4.1.2 => 4.1.2
  vite: ^6.3.2 => 6.3.2
```

### Used Package Manager

npm

### Logs

<details>
<summary>Click to expand!</summary>

```shell
$ npx vue-tsc --noEmit

Error: vite.config.ts(7,8): error TS1192: Module '"/home/runner/work/gdqreminder-mobile/gdqreminder-mobile/node_modules/@vitejs/plugin-legacy/dist/index"' has no default export.
Error: Process completed with exit code 2.
```
</details>

### Validations

- [x] Follow our [Code of Conduct](https://github.com/vitejs/vite/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md).
- [x] Read the [docs](https://vite.dev/guide).
- [x] Check that there isn't [already an issue](https://github.com/vitejs/vite/issues) that reports the same bug to avoid creating a duplicate.
- [x] Make sure this is a Vite issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to [vuejs/core](https://github.com/vuejs/core) instead.
- [x] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitejs/vite/discussions) or join our [Discord Chat Server](https://chat.vite.dev/).
- [x] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/plugin-legacy/package.json
+++ b/packages/plugin-legacy/package.json
@@ -58,7 +58,7 @@
   "devDependencies": {
     "acorn": "^8.14.1",
     "picocolors": "^1.1.1",
-    "unbuild": "^3.5.0",
+    "unbuild": "3.4.2",
     "vite": "workspace:*"
   }
 }

